src/clipboard_handlers/borders_clipboard.ts

```diff
@@ -1,43 +1,59 @@
-import { positionToZone, recomputeZones } from "../helpers";
+import { splitZoneForPaste } from "../helpers/clipboard/clipboard_helpers";
+import { groupConsecutive } from "../helpers/misc";
+import { ZoneBorder } from "../plugins/core";
 import {
-  Border,
-  CellPosition,
   ClipboardCellData,
   ClipboardOptions,
   ClipboardPasteTarget,
   HeaderIndex,
   UID,
-  Zone,
 } from "../types";
 import { AbstractCellClipboardHandler } from "./abstract_cell_clipboard_handler";
 
 type ClipboardContent = {
-  borders: (Border | null)[][];
+  borders: ZoneBorder[];
+  width: number;
+  height: number;
 };
 
 export class BorderClipboardHandler extends AbstractCellClipboardHandler<
   ClipboardContent,
-  Border | null
+  ZoneBorder
 > {
-  private queuedBordersToAdd: Record<string, Zone[]> = {};
-
   copy(data: ClipboardCellData): ClipboardContent | undefined {
     const sheetId = data.sheetId;
     if (data.zones.length === 0) {
       return;
     }
-    const { rowsIndexes, columnsIndexes } = data;
-    const borders: (Border | null)[][] = [];
-
-    for (const row of rowsIndexes) {
-      const bordersInRow: (Border | null)[] = [];
-      for (const col of columnsIndexes) {
-        const position = { col, row, sheetId };
-        bordersInRow.push(this.getters.getCellBorder(position));
+    const borders: ZoneBorder[] = [];
+    let colsBefore = 0;
+    for (const cols of groupConsecutive(data.columnsIndexes)) {
+      let rowsBefore = 0;
+      for (const rows of groupConsecutive(data.rowsIndexes)) {
+        const zone = {
+          left: cols[0],
+          right: cols[cols.length - 1],
+          top: rows[0],
+          bottom: rows[rows.length - 1],
+        };
+        borders.push(
+          ...this.getters.getBorders(sheetId, zone).map((zb) => {
```

Comment:
I don't know what `zb` is when reading
```suggestion
          ...this.getters.getBorders(sheetId, zone).map(({ style, zone: borderZone}) => {
```